interview-questions.md

```diff
@@ -8,62 +8,68 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The .map() method takes in 3 parameters. Only one is required. The first parameter is value. I can't remember the second and third.
 
-  Researched answer:
+  Researched answer: The map method take 3 parameters which are (value, index, array). From what I understand, value is the only required parameter, and holds the value of the current element. Index is an optional parameter and is suppossed to hold the index of the current element. Array is also optional, and is used to hold the array. Some sites have the syntax as the following... 
+  
+  array.map(function(currentValue, index, arr), thisValue)
+  
+  They list thisValue as another optional parameter thats used to hold the value that's passed into the function. .map() is a non-mutating method, otherwise known as a getter method, also known as an accessor method.
 
 
 
 3. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is writing code in a more concise way resulting in less code with the same results.
 
-  Researched answer:
+  Researched answer: Object destructuring is a Javascript feature that lets you take out properties from objects and put them into variables. A clearer way of saying that is it lets you 'extract' properties from objects and 'bind' them into variables. You can extract multiple properties using a single statement. Before object destructuring existed in Javascript (pre-ES2015), you would have to declare each object property with var. If you had 10 properties, you would need 10 var declarations. With destructuring, you can put them all in a single statement in one line of code.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is a block of code that's created to perform a specific action. A method is a built-in function of a language that's used within other functions.
 
-  Researched answer:
+  Researched answer: A function is a block of code thats declared by name and passed paramenters that can return data (if needed). A method is almost identical to a function, except a method is paired with an object. For example, the .lowercase() method needs to be called on a variable. You can't just call .lowerCase() by itself. But if I created a function called lowerCase(), I could simply call that function on it's own and it would work.
```

Comment:
Great description of a difference. Methods belong to classes so we can call any method on something of the same class! String methods on strings, custom methods on a custom class instance.